actionpack/lib/action_dispatch/http/content_security_policy.rb

```diff
@@ -20,25 +20,37 @@ def call(env)
         return response unless html_response?(headers)
         return response if policy_present?(headers)
 
-        if policy = request.content_security_policy
+        policy = request.content_security_policy
+        report_only_policy = determine_report_only_policy(request)
+
+        if policy || report_only_policy
           nonce = request.content_security_policy_nonce
           nonce_directives = request.content_security_policy_nonce_directives
           context = request.controller_instance || request
-          headers[header_name(request)] = policy.build(context, nonce, nonce_directives)
+
+          headers[header_name(request)] = policy.build(context, nonce, nonce_directives) if policy
+
+          headers[POLICY_REPORT_ONLY] = report_only_policy.build(context, nonce, nonce_directives) if report_only_policy
         end
 
         response
       end
 
       private
+        def determine_report_only_policy(request)
+          return if request.content_security_policy_report_only.is_a?(TrueClass)
```

Comment:
this is to have backward support for current DSL
```
Rails.application.config.content_security_policy do |p|
   ...
end

Rails.application.config.content_security_policy_report_only = true
```

Dropping support for this would simplify the implementation quite a bit, but would be a breaking change.

Reply:
This checking for `TrueClass` doesn't feel right to me - I think there's a cleaner way to do this but will need to study it a bit more

Reply:
Please let me know if you can think of anything. This is the first time I can recall needing to do this. We want to assure the value is actually true and not just truthy.

Reply:
I think I'd have used `.eql?(true)` or `== true`, but `is_a?(TrueClass)` is clear in its specificity. 👍